syntax = "proto3";

import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";

option go_package = "proto/article";

message TopicArticles {
        map<string, Articles> topicArticles = 1;
}

message Articles {
        repeated Article articles = 1;
}

message Article {
        string title = 1;
        string description = 2;
        string url = 3;
        string imageUrl = 4;
        string source = 5;
        google.protobuf.Timestamp date = 6;
}

service ArticleService {
        rpc GetTopicArticles(GetTopicArticlesRequest) returns (TopicArticles);
        rpc SetTopicArticles(SetTopicArticlesRequest) returns (google.protobuf.Empty);
}

message SetTopicArticlesRequest {
        map<string, Articles> topicArticles = 1;
}

message GetTopicArticlesRequest {
        repeated string topics = 1;
}

